src/js/base/Icon.js

```diff
@@ -31,17 +33,18 @@ export const Icon = ({ hoverable, style, ...props }) => {
     const handleClick = useCallback(props.onClick, [props.onClick]);
 
     const className = `${props.className ? props.className + " " : ""} ${props.faStyle} fa-${props.name}`;
-
+    const renderAs = !props.onClick ? "i" : "button";
```

Comment:
Don't think you need to declare this variable. Just provide prop directly in component call.